pkg/sessiontxn/isolation/base.go

```diff
@@ -468,6 +445,159 @@ func (p *baseTxnContextProvider) getSnapshotByTS(snapshotTS uint64) (kv.Snapshot
 	return snapshot, nil
 }
 
+func (p *baseTxnContextProvider) SetOptionsAfterBegin(txn kv.Transaction) {
+	sessVars := p.sctx.GetSessionVars()
+
```

Comment:
Need to check the txn is in execution phase, report error if not to avoid mis-usages.

Reply:
Added an assertion in test.